hw/xbox/mcpx_apu.c

```diff
@@ -811,11 +1126,404 @@ static const MemoryRegionOps ep_ops = {
+      voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLC, NV_PAVS_VOICE_TAR_VOLC_VOLUME4),
+      voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLC, NV_PAVS_VOICE_TAR_VOLC_VOLUME5),
+      (voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLC, NV_PAVS_VOICE_TAR_VOLC_VOLUME6_B11_8) << 8) |
+      (voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLB, NV_PAVS_VOICE_TAR_VOLB_VOLUME6_B7_4) << 4) |
+      voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLA, NV_PAVS_VOICE_TAR_VOLA_VOLUME6_B3_0),
+      (voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLC, NV_PAVS_VOICE_TAR_VOLC_VOLUME7_B11_8) << 8) |
+      (voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLB, NV_PAVS_VOICE_TAR_VOLB_VOLUME7_B7_4) << 4) |
+      voice_get_mask(d, v, NV_PAVS_VOICE_TAR_VOLA, NV_PAVS_VOICE_TAR_VOLA_VOLUME7_B3_0),
+    };
+
+    if (v == 0x0044) {
+      static unsigned int x = 0;
+      
+      for(unsigned int i = 0; i < 0x20; i++) {
+        uint32_t sample_pos = cbo + (uint32_t)(i * rate * overdrive);
+        //uint32_t sample_pos = cbo + (uint32_t)(i * rate * overdrive);
+        //FIXME: The mod ebo thing is a hack!
+        uint32_t linear_addr = ba + (sample_pos % (ebo + 1)) * container_size;
+        hwaddr addr = get_data_ptr(d->regs[NV_PAPU_VPSGEADDR], 0xFFFFFFFF, linear_addr);
+        //printf("Sampling from 0x%08X\n", addr);
+        int16_t sample = (int16_t)lduw_le_phys(addr);
+        wav_out_write(&buf_wav_out[0], &sample, 2); // voice input
+        x++;
+      }
+      wav_out_update(&buf_wav_out[0]);
+
+      for(unsigned int i = 0; i < 0x20; i++) {
+        wav_out_write(&buf_wav_out[1], &samples[0][i], 3); // mixbuf for voice
+        wav_out_write(&buf_wav_out[1], &samples[channels - 1][i], 3); // mixbuf for voice
+      }
+      wav_out_update(&buf_wav_out[1]);
+    }
+
+    if (paused) {
+      assert(sizeof(samples) == 0x20 * 4 * 2);
+      memset(samples, 0x00, sizeof(samples));
+    } else {
+      cbo += 0x20 * rate * overdrive;
+      voice_set_mask(d, v, NV_PAVS_VOICE_PAR_OFFSET, NV_PAVS_VOICE_PAR_OFFSET_CBO, cbo);
+    }
+
+    const char* name = "unk";
+#if 0
+    // Whitelisted: !
+    if (ba == 0x1F08) {
+      name = "Thunder";
+    } else if (ba == 0x141E8) { // White noise (in bad range?! 16U?!)
+      name = "White Noise";
+    } else if (ba == 0xD1F0) { // Beep sound
+      name = "Beep";
+    } else if (ba == 0x102E8) { // Flubber / Bubble sound
+      name = "Flubber";
+
+    // Blacklisted: !
+    } else if (ba == 0xF998) { // garbage? some sawtooth I guess
+      name = "Sawtooth";
+      goto skipvoice;
+    } else if (ba == 0x978) { // silence
+      name = "Silence";
+      goto skipvoice;
+
+
+    ///// Wavebank XDK sample
+    } else if (ba == 0x246C50) { // garbage? some sawtooth I guess
+      name = "Sound 9"; // "The call is tails"
+    } else if (ba == 0x3A6694) { // garbage? some sawtooth I guess
+      name = "Sound 13"; // Music
+
+
+    ///// Catchall
+    } else {
+      printf("Skipping ", ba);
+      goto skipvoice;
+    }
+    
+
+#endif
```

Comment:
This block should never have landed upstream.
This was for debugging the Xbox boot animation (assigning labels to sounds).